test/step_definitions/hooks.js

```diff
@@ -0,0 +1,9 @@
+const { After } = require('@cucumber/cucumber');
+const { browser } = require('protractor');
+
+
+After(async function () {
+    const screenshot = await browser.takeScreenshot();
+    const decodedImage = new Buffer.from(screenshot, 'base64');
+    return this.attach(decodedImage, 'image/png'); 
+});
```

Comment:
Please add a hook to attach screenshot if the step fails.

Reply:
improved